### PR TITLE
Lenovo finetune modules

### PR DESCRIPTION
**Pull request by amuraleedhar**
##### SUMMARY
In this PR I am making some fine tuning to arguments supplied to module. After moving to persietnce connection from paramiko, few module parameters have become non manadatory. Some fine tuning has to be done in documentation too. Added Bacon Switch (NE0152T) also into device rule file.

ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
lib\ansible\modules\network\cnos\cnos_backup.py
lib\ansible\modules\network\cnos\cnos_bgp.py
lib\ansible\modules\network\cnos\cnos_conditional_command.py
lib\ansible\modules\network\cnos\cnos_conditional_template.py
lib\ansible\modules\network\cnos\cnos_config.py
lib\ansible\modules\network\cnos\cnos_factory.py
lib\ansible\modules\network\cnos\cnos_facts.py
lib\ansible\modules\network\cnos\cnos_image.py
lib\ansible\modules\network\cnos\cnos_interface.py
lib\ansible\modules\network\cnos\cnos_portchannel.py
lib\ansible\modules\network\cnos\cnos_reload.py
lib\ansible\modules\network\cnos\cnos_save.py
lib\ansible\modules\network\cnos\cnos_showrun.py
lib\ansible\modules\network\cnos\cnos_rollback.py
lib\ansible\modules\network\cnos\cnos_template.py
lib\ansible\modules\network\cnos\cnos_vlag.py
lib\ansible\modules\network\cnos\cnos_vlan.py
lib\ansible\modules\network\enos\enos_config.py
lib\ansible\modules\network\enos\enos_facts.py
lib\ansible\module_utils\network\cnos\cnos_devicerules.py

##### ANSIBLE VERSION
ansible 2.8.0.dev0 (devel 960dfa862b) last updated 2018/10/08 17:48:02 (GMT +550)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/ansible/sheru/ansible/lib/ansible
  executable location = /home/ansible/sheru/ansible/bin/ansible
  python version = 2.7.6 (default, Nov 23 2017, 15:49:48) [GCC 4.8.4]

##### ADDITIONAL INFORMATION
None.

### CODE REVIEW

File: lib/ansible/modules/network/cnos/cnos_interface.py [hunk 1]
```diff
@@ -379,9 +305,6 @@
 
 - name: Test Interface Ethernet - spanning tree 2
   cnos_interface:
-      host: "{{ inventory_hostname }}"
-      username: "{{ hostvars[inventory_hostname]['ansible_ssh_user'] }}"
-      password: "{{ hostvars[inventory_hostname]['ansible_ssh_pass'] }}"
       deviceType: "{{ hostvars[inventory_hostname]['deviceType'] }}"
       enablePassword: "{{ hostvars[inventory_hostname]['enablePassword'] }}"
```
Comment (Qalthos) [inline lib/ansible/modules/network/cnos/cnos_interface.py]: You probably meant to remove `enablePassword` here?

File: lib/ansible/modules/network/enos/enos_config.py [hunk 2]
```diff
@@ -38,7 +38,7 @@
     a deterministic way.
 extends_documentation_fragment: enos
 notes:
-  - Tested against ENOS 8.4.1.2
+  - Tested against ENOS 8.4.1.0
```
Comment (Qalthos) [inline lib/ansible/modules/network/enos/enos_config.py]: Just making a note here as this is the only one left that has a fourth digit (`8.4.1.0` vs `8.4.1`)

**Review by Qalthos — COMMENTED** [review]
Looks good in general. I only found a few things that looked like they might be oversights.

File: lib/ansible/modules/network/cnos/cnos_interface.py (hunk 1 above)
Comment (amuraleedhar) [reply lib/ansible/modules/network/cnos/cnos_interface.py]: Got missed out. Changing now

File: lib/ansible/modules/network/enos/enos_config.py (hunk 2 above)
Comment (amuraleedhar) [reply lib/ansible/modules/network/enos/enos_config.py]: Changing now

**Pull request merged**